chartfx-chart/src/main/java/de/gsi/chart/renderer/spi/ErrorDataSetRenderer.java

```diff
@@ -169,6 +174,11 @@ public Marker getMarker() {
             stopStamp = ProcessingProfiler.getTimeStamp();
             final DataSet dataSet = localDataSetList.get(dataSetIndex);
 
+            if(!dataSet.isVisible()) {
+                localDataSetList.remove(dataSetIndex); // TODO (ennerf): is this correct and/or necessary?
```

Comment:
Since the default colour and styles are coupled to the data set index in the list, I'd just add this as a continue statement in the main loop, e.g.

```java
for (int dataSetIndex = localDataSetList.size() - 1; dataSetIndex >= 0; dataSetIndex--) {
    // [..]
    if(!dataSet.isVisible()) {
        continue;
    }
    // [..]
}
```
This way muted DataSets wouldn't change the colour and/or other styles of the other non-muted DataSets.

Reply:
Done. I got rid of some of the `drawnDataSets = new ArrayList(localDataSetList)` copies where every dataSet gets added anyways.